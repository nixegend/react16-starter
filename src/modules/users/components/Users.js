import React from 'react';

class Users extends React.Component {
  componentWillMount() {
    this.props.loadUsersList();
  }

  render() {
    const { users } = this.props;

    return (
      users.length ?
        <div className="container">
          <ul className="list">
            {
              users.map(user => (
                <li key={user.id}>
                  <span>{user.name}</span>
                  <button type="button" onClick={() => this.props.removeUser(user.id)}>remove</button>
                </li>
              ))
            }
          </ul>
          <img src="img/wb.png" alt="img" />
        </div>
        : null
    );
  }
}

export default Users;
