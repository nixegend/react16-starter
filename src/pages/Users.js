import React from 'react';
import UsersContainer from '../modules/users/containers/UsersContainer';

const Users = () => (
  <UsersContainer />
);

export default Users;
