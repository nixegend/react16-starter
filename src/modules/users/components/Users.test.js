import React from 'react';
import renderer from 'react-test-renderer';
import Users from './Users';

const props = {
  removeUser: () => undefined,
  loadUsersList: () => undefined,
  users: [
    { name: 'Bob3', id: 'sfdsdfsdf' },
    { name: 'Bob2', id: 'dfgwerwrd' },
    { name: 'Bob1', id: 's234sfsdg' },
  ],
};

describe('<Users />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Users {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
