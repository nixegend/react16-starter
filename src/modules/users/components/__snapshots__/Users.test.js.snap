// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Users /> renders correctly 1`] = `
<div
  className="container"
>
  <ul
    className="list"
  >
    <li>
      <span>
        Bob3
      </span>
      <button
        onClick={[Function]}
        type="button"
      >
        remove
      </button>
    </li>
    <li>
      <span>
        Bob2
      </span>
      <button
        onClick={[Function]}
        type="button"
      >
        remove
      </button>
    </li>
    <li>
      <span>
        Bob1
      </span>
      <button
        onClick={[Function]}
        type="button"
      >
        remove
      </button>
    </li>
  </ul>
  <img
    alt="img"
    src="img/wb.png"
  />
</div>
`;
